qiskit/quantum_info/synthesis/one_qubit_decompose.py

```diff
@@ -14,233 +14,393 @@
 from qiskit.quantum_info.operators.predicates import is_unitary_matrix
 
-DEFAULT_ATOL = 1e-12
+DEFAULT_SIMPLIFY_TOLERANCE = 1e-12
+DEFAULT_ATOL = 1e-7
 
 
 class OneQubitEulerDecomposer:
-    """A class for decomposing 1-qubit unitaries into Euler angle rotations.
-
-    Allowed basis and their decompositions are:
-        U3: U -> exp(1j*phase) * U3(theta, phi, lam)
-        U1X: U -> exp(1j*phase) * U1(lam).RX(pi/2).U1(theta+pi).RX(pi/2).U1(phi+pi)
-        ZYZ: U -> exp(1j*phase) * RZ(phi).RY(theta).RZ(lam)
-        ZXZ: U -> exp(1j*phase) * RZ(phi).RX(theta).RZ(lam)
-        XYX: U -> exp(1j*phase) * RX(phi).RY(theta).RX(lam)
+    r"""A class for decomposing 1-qubit unitaries into Euler angle rotations.
+
+    The resulting decomposition is parameterized by 3 Euler rotation angle
+    parameters :math:`(\theta, \phi\ lambda)`, and a phase parameter
+    :math:`\gamma`. The value of the parameters for an input unitary depends
+    on the decomposition basis. Allowed bases and the resulting circuits are
+    shown in the following table. Note that for the non-Euler bases (U3, U1X,
+    RR), the ZYZ euler parameters are used.
+
+    .. list-table:: Supported circuit bases
+        :widths: auto
+        :header-rows: 1
+
+        * - Basis
+          - Euler Angle Basis
+          - Decomposition Circuit
+        * - 'ZYZ'
+          - :math:`Z(\phi) Y(\theta) Z(\lambda)`
+          - :math:`e^{i\gamma} R_Z(\phi).R_Y(\theta).R_Z(\lambda)`
+        * - 'ZXZ'
+          - :math:`Z(\phi) X(\theta) Z(\lambda)`
+          - :math:`e^{i\gamma} R_Z(\phi).R_X(\theta).R_Z(\lambda)`
+        * - 'XYX'
+          - :math:`X(\phi) Y(\theta) X(\lambda)`
+          - :math:`e^{i\gamma} R_X(\phi).R_Y(\theta).R_X(\lambda)`
+        * - 'U3'
+          - :math:`Z(\phi) Y(\theta) Z(\lambda)`
+          - :math:`e^{i\gamma}{2}\right)\right)} U_3(\theta,\phi,\lambda)`
+        * - 'U1X'
+          - :math:`Z(\phi) Y(\theta) Z(\lambda)`
+          - :math:`e^{i \gamma}{2}\right)\right)}
+            :math:`U_1(\phi+\pi).R_X\left(\frac{\pi}{2}\right).U_1(\theta+\pi).`
+            :math:`R_X\left(\frac{\pi}{2}\right).U_1(\lambda)`
+        * - 'RR'
+          - :math:`Z(\phi) Y(\theta) Z(\lambda)`
+          - :math:`e^{i\gamma} R\left(-\pi,\frac{\phi-\lambda+\pi}{2}\right).`
+            :math:`R\left(\theta+\pi,\frac{\pi}{2}-\lambda\right)`
     """
     def __init__(self, basis='U3'):
-        if basis not in ['U3', 'U1X', 'ZYZ', 'ZXZ', 'XYX']:
-            raise QiskitError("OneQubitEulerDecomposer: unsupported basis")
-        self._basis = basis
+        """Initialize decomposer
+
+        Supported bases are: 'U3', 'U1X', 'RR', 'ZYZ', 'ZXZ', 'XYX'.
+
+        Args:
+            basis (str): the decomposition basis [Default: 'U3]
+
+        Raises:
+            QiskitError: If input basis is not recognized.
+        """
+        # Default values
+        self._basis = 'U3'
+        self._params = self._params_u3
+        self._circuit = self._circuit_u3
```

Comment:
'U3' value is already the kwarg default. so it should not be repeated here I think, and you should use the given `basis` to initialize these properties.

Reply:
This was mostly just to make sure internal attributes were declared __init__ method, rather than being hidden in the `basis` property setter. Is there a "proper" way to do this?